drivers/i2c/i2c_mcux.c

```diff
@@ -186,15 +186,15 @@ static const struct i2c_driver_api i2c_mcux_driver_api = {
 static void i2c_mcux_config_func_0(struct device *dev);
 
 static const struct i2c_mcux_config i2c_mcux_config_0 = {
-	.base = I2C0,
+	.base = (void *)CONFIG_I2C_MCUX_0_BASE_ADDRESS,
```

Comment:
Should cast to (I2C_Type *) instead of (void *). And if you're going to switch to the dts base address here, you should also do it for instance 1.